### ISSUE TITLE
fix: resolve detached thread scope warning in PinterestOAuth (SonarCloud cpp:S5962)

### ISSUE THREAD

**Issue by spiritEcosse** [spiritEcosse/foxy_server#89]
## Problem

SonarCloud flags `std::jthread(...).detach()` in `PinterestOAuth::callback` as having too narrow a scope (rule cpp:S5962, MINOR).

\`\`\`cpp
std::jthread([callbackPtr, code]() {
    exchangeTokenAndSave(callbackPtr, code);
}).detach();
\`\`\`

## Context

The detach is required because `PinterestOAuth::callback` runs on the Drogon event loop thread and must return immediately. The blocking CPR HTTP call to Pinterest's token endpoint cannot run on the event loop, so it is offloaded to a detached thread.

Giving the thread a "bigger scope" (e.g. storing it as a class member) is not straightforward because the controller is used concurrently across many requests.

## Options to investigate

1. Use Drogon's built-in async facilities (e.g. `drogon::async_run` or a thread pool) instead of a raw `jthread`
2. Use a thread pool (e.g. `drogon::app().getThreadPool()`) to queue the blocking work
3. Add a SonarCloud rule exclusion for this specific pattern
